src/time/span.cr

```diff
@@ -60,13 +60,24 @@ struct Time::Span
   def self.new(days : Int, hours : Int, minutes : Int, seconds : Int, nanoseconds : Int = 0)
     new(
       seconds: compute_seconds!(days, hours, minutes, seconds),
-      nanoseconds: nanoseconds.to_i64,
+      nanoseconds: nanoseconds,
     )
   end
 
   def initialize(*, seconds : Int, nanoseconds : Int)
+    # check for possible overflow
+    # seconds could be too big
+    unless Int64::MIN <= seconds <= Int64::MAX
+      raise ArgumentError.new "Time::Span too big or too small"
+    end
```

Comment:
Does a negative Time::Span make sense?

Reply:
Sure, why wouldn't it? If you subtract an earlier instance from a later one, the difference is a negative duration.

Reply:
I think it does. Otherwise some operations are a bit hard to do.